File: 0398-random-pick-index/0398-random-pick-index.py
class Solution:

    def __init__(self, nums: List[int]):
        self.hash_map = {}

        for index, item in enumerate(nums):
            if item not in self.hash_map:
                self.hash_map[item] = []
            self.hash_map[item].append(index)
        print(self.hash_map)
        self.total = len(nums)-1

    def pick(self, target: int) -> int:
        index_list = self.hash_map[target]
        random_num = random.randint(0, len(index_list)-1)

        return index_list[random_num]
    
# Your Solution object will be instantiated and called as such:
# obj = Solution(nums)
# param_1 = obj.pick(target)

File: 0398-random-pick-index/0398-random-pick-index.cpp
class Solution {
private:
    unordered_map<int,vector<int>>valtoIndex;
public:
    Solution(vector<int>& nums) {
        for(int i=0;i<nums.size();i++){
            valtoIndex[nums[i]].push_back(i);
        }
    }
    
    int pick(int target) {
        return valtoIndex[target][rand() % valtoIndex[target].size()];
    }
};

/**
 * Your Solution object will be instantiated and called as such:
 * Solution* obj = new Solution(nums);
 * int param_1 = obj->pick(target);
 */